fix: disable edit button on App Details Summary and Parameters tabs multi-source apps

ishitasequeira opened the pull request:
fix: disable edit button on App Details Summary and Parameters tabs multi-source apps

fixes #11757 & #11758

* Summary Tab
![image](https://user-images.githubusercontent.com/46771830/216404263-febe5165-493b-4ca5-ab71-cb63a0e12e65.png)


* Parameters Tab
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/46771830/213520974-c74351d1-8d1c-4a19-9582-9b9018349895.png">


Signed-off-by: ishitasequeira <ishiseq29@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

crenshaw-dev reviewed (changes requested):
I think instead of disabling all edits in the App Summary page, we should only disable the fields which are related to the `sources` block. For example, editing the app's labels shouldn't be disabled.

ishitasequeira commented:
> I think instead of disabling all edits in the App Summary page, we should only disable the fields which are related to the `sources` block. For example, editing the app's labels shouldn't be disabled.



> I think instead of disabling all edits in the App Summary page, we should only disable the fields which are related to the `sources` block. For example, editing the app's labels shouldn't be disabled.

I have addressed the comment.
